### ISSUE TITLE
[feat] report-unused-disable-directives

### ISSUE THREAD

**Issue by loynoir** [rust-lang/rust-clippy#14179]
### What it does

Similar to eslint

https://eslint.org/docs/latest/use/command-line-interface#--report-unused-disable-directives

> --report-unused-disable-directives
>
> This option causes ESLint to report directive comments like // eslint-disable-line when no errors would have been reported on that line anyway.

When dev, may suppress some rules.

```rs
#[allow(XXX)]
dev_code_block_warn_xxx
```

Then, at another time point, no need to suppress.

But, the unused suppress is not removed.

```rs
#[allow(XXX)]
dev_code_block_without_xxx
```

### Advantage

- clean code, if code without XXX problem, no need to `#[allow(XXX)]`

### Example

Warn

```rs
#[allow(XXX)]
dev_code_block_without_xxx
```

**Comment by samueltardieu** [comment]
Note that you can use the `allow_attribute` lint to forbid the `allow` attribute, then use `expect` instead of `allow`. `expect` will warn if the triggering condition is no longer valid.
